src/base/memory_monitor_sr.cpp

```diff
@@ -20,18 +20,62 @@
  * memory_monitor_sr.cpp - Implementation of memory monitor module
  */
 
+#include <malloc.h>
+#include <cassert>
 #include <cstring>
 #include <algorithm>
-#include <cassert>
 
 #include "memory_monitor_sr.hpp"
 
+#ifndef HAVE_USR_INCLUDE_MALLOC_H
+#define HAVE_USR_INCLUDE_MALLOC_H
+#endif
+
+typedef struct mmon_metainfo
+{
+  uint64_t size;
+  int tag_id;
+  int magic_number;
+} MMON_METAINFO;
+
 namespace cubmem
 {
+  memory_monitor *mmon_Gl = nullptr;
+
   memory_monitor::memory_monitor (const char *server_name)
-    : m_server_name {server_name} {}
+    : m_server_name {server_name}
+  {
+    std::string magic_string ("MMON");
+    memcpy (&magic, magic_string.c_str (), sizeof (int));
+  }
+
+  size_t memory_monitor::get_alloc_size (const char *ptr)
+  {
+    size_t ret;
```

Comment:
How about reuse `alloc_size` local variable? The `ret` is not good readability.

Reply:
I understand it. Done.